pack.go

```diff
@@ -128,62 +187,26 @@ func Pack(ctx context.Context, pusher content.Pusher, layers []ocispec.Descripto
 	return manifestDesc, nil
 }
 
-// PackArtifact packs the given blobs, generates an ORAS Artifact Manifest for
-// the pack, and pushes it to a content storage.
-// If succeeded, returns a descriptor of the manifest.
-// Returns ErrMissingArtifactType if artifactType is empty.
-// Reference: https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md
-func PackArtifact(ctx context.Context, pusher content.Pusher, artifactType string, blobs []artifactspec.Descriptor, opts PackArtifactOptions) (ocispec.Descriptor, error) {
-	if artifactType == "" {
-		// artifactType is required for ORAS Artifact Manifest
-		return ocispec.Descriptor{}, ErrMissingArtifactType
-	}
-
-	if createdTime, ok := opts.ManifestAnnotations[artifactspec.AnnotationArtifactCreated]; ok {
-		// if AnnotationArtifactCreated is provided, validate its format
+// ensureAnnotationCreated ensures that annotationCreatedKey is in annotations,
+// and that its value conforms to RFC 3339. Otherwise returns a new annotation
+// map with annotationCreatedKey created.
+func ensureAnnotationCreated(annotations map[string]string, annotationCreatedKey string) (map[string]string, error) {
```

Comment:
The creation time annotation is not mandatory in the [spec](https://github.com/opencontainers/image-spec/blob/main/artifact.md), but I think it's good to add it as a built-in annotation of oras-go.